### PR TITLE
feat(core): Add metrics for queue mode

### PR DESCRIPTION
**Pull request by netroy**
## TODO
- [ ] Fix webhooks in queue mode when not using "respond immediately"

Screenshot from 2023-09-26 17-27-31

### CODE REVIEW

File: packages/cli/src/Queue.ts [hunk 1]
```diff
@@ -59,16 +64,44 @@ export class Queue {
...
+					if (progress === -1) {
+						// Job has to get canceled
+						this.emit('job:cancelled', jobId);
+					}
+				} else {
+					const { executionId, response } = progress as WebhookResponse;
+					this.activeExecutions.resolveResponsePromise(
+						executionId,
+						WebhookHelpers.decodeWebhookResponse(response),
+					);
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/Queue.ts:87]: yeah. I think that's the same as the `TODO` in the description.
I suspect it's the refactoring that broke that behavior. We might need to add some unit tests for webhooks before we visit this PR again.

**Comment by Dasio** [comment]
Hello, is there plan to continue on this feature? it would help us with observability

**Comment by Joffcom** [comment]
The PR is still open so at the moment the plan would be to finish this at some point but we have no timeframe as other features / issues have a higher priority.

File: packages/cli/src/Queue.ts [hunk 2]
```diff
@@ -60,51 +65,62 @@ export class Queue {
...
 
-		this.jobQueue.on('global:progress', (jobId, progress: WebhookResponse) => {
-			this.activeExecutions.resolveResponsePromise(
-				progress.executionId,
-				decodeWebhookResponse(progress.response),
-			);
-		});
-	}
+		const instanceType = config.getEnv('generic.instanceType');
+		if (instanceType === 'main') {
```
Comment (krynble) [inline packages/cli/src/Queue.ts:69]: Now with multi-main we might get duplicate counts if there are multiple main instances listening to this, so this code should subescribe to the `leadershipChange` event and only fire if current instance is leader

**Pull request closed without merging**